daemon: Ensure buildkit created container's isolation mode is consistent with daemon's config in Windows

Vigilans opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

Currently, when enabling buildkit in Windows, the intermediate container created by buildlkit when building image will not follow docker daemon's configuration but stick to `process` isolation. This will cause bug in machines that does not support process isolation, making them unable to build images with buildkit.

This PR fixes this issue to ensure buildkit created container's isolation mode consistent with daemon's config.

**- How I did it**

1. Add `SpecOpts` to `ExecutorOption` (Done in buildkit's repository: https://github.com/moby/buildkit/pull/6206, superseded by https://github.com/moby/buildkit/pull/6224)
2. Expose `DefaultIsolation` field of docker daemon
3. Add `WithWindowsHyperV` specOpts when daemon's default isolation mode is `hyperv`.

**- How to verify it**

1. Follow this setup to enable buildkit in windows: https://gist.github.com/profnandaa/9ad40dbd90ebad99896ee0fed2916406
2. In a machine that does not support process isolation mode, run `docker buildx build .`
3. Will see errors like:
```
ERROR: failed to build: failed to solve: process "cmd /S /C echo \"hello wolrd"\" did not complete successfully: failed to create shim task: hcs:CreateComputeSystem hcdqcsa2mflizmrzrt2z9yox: 
The container operating system does not match the host operating system.
```

**- Human readable description for the release notes**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
It must be placed inside the below triple backticks section.

NOTE: Only fill this section if changes introduced in this PR are user-facing.
The PR must have a relevant impact/ label.
-->
```markdown changelog
Windows: Fix BuildKit creating containers which isolation mode is inconsistent with the daemon's config
```

**- A picture of a cute animal (not mandatory but encouraged)**

<img width="185" height="265" alt="yasha" src="https://github.com/user-attachments/assets/ede49bf9-a3dc-4b7c-8d1f-008d01c238ef" />

vvoland commented on daemon/internal/builder-next/executor_linux.go:
```diff
@@ -6,6 +6,7 @@ import (
 	"path/filepath"
 	"strconv"
 
+	containerdoci "github.com/containerd/containerd/v2/pkg/oci"
```
```suggestion
	coci "github.com/containerd/containerd/v2/pkg/oci"
```

We tend to use the `coci` alias for this one

vvoland reviewed (changes requested):
LGTM, thanks! (but will leave a blocking review until the Buildkit changes are vendored)

Vigilans committed:
daemon: Ensure buildkit created container's isolation mode consistent with daemon's config

- Introduced DefaultIsolation method in the Daemon to return the daemon configured isolation mode for Windows.

Signed-off-by: Vigilans <vigilans@foxmail.com>

vvoland reviewed (approved):
LGTM
